### PR TITLE
Ansible user management

### PR DESCRIPTION
**Pull request by qburst-arjunm**
# Overview

This includes Ansible role to create/manage users in various tools like AWS, Linux, Jenkins, MySQL and PostgreSQL.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,4 +11,5 @@ To facilitate easy navigation and access to the resources within this repository
...
 2. [**GitHub Actions**](/github-actions/README.md):  This folder contains a selection of GitHub Actions that have been designed to optimize workflows. These actions offer a reusable and configurable approach for achieving efficiency and consistency in your development practices.
-3. [**Terraform**](/terraform/README.md): In this folder, you will discover a comprehensive set of Terraform modules created to address common use cases. These modules helps you to rapidly provision and manage infrastructure resources with ease, leveraging the power and flexibility of Terraform.
\ No newline at end of file
+3. [**Terraform**](/terraform/README.md): In this folder, you will discover a comprehensive set of Terraform modules created to address common use cases. These modules helps you to rapidly provision and manage infrastructure resources with ease, leveraging the power and flexibility of Terraform.
+4. [**Ansible**](/ansible/README.md): In this folder, you will discover a comprehensive set of ansible roles to create/manage users. This role helps you to easily manage users in different platforms like AWS, Linux, Jenkins, MySQL and Postgres.
```
Comment (qburst-praven) [inline README.md:15]: In this folder, you will discover a comprehensive set of Ansible roles to run many common scenarios.

File: ansible/user-management/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+The same is provided in the main.yml residing outside the role. You can use the following command to run all the tasks.
+
+### ansible-playbook main.yml
+
+You can use --skip-tags to exclude any particular role
+
+### Ex: ansible-playbook main.yml --skip-tags linux
+
+# License
```
Comment (qburst-praven) [inline ansible/user-management/README.md:71]: Please remove the License area

File: ansible/README.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+This is where the various roles resides.
+
+## Prerequisites
+- Ansible
+- Python
+
+## Role List
+### 1. User-management 
+[User management in various tools](/ansible/user-management/main.yml)
+
```
Comment (qburst-praven) [inline ansible/README.md:15]: This role helps you to manage/creates users in different platforms like AWS, Linux, Jenkins, MySQL and Postgres.

File: ansible/user-management/tasks/linux-add-user.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+- debug:
+    var: user_check
+
+- name: Creating new user in Linux
+  user:
+    name: "{{ users.linux.username }}"
+    state: present
+    groups: "{{ users.linux.groups }}"
+    password: "{{ users.linux.password }}"
```
Comment (qburst-praven) [inline ansible/user-management/tasks/linux-add-user.yml:16]: pls dont use password based login for Linux users. Key based is preferred. You can keep an option to use password if provided else to set it to null.

File: ansible/user-management/tasks/linux-add-user.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  when: "ansible_os_family == 'RedHat' or ansible_os_family == 'Debian'"
+
+- debug:
+    var: user_check
+
+- name: Creating new user in Linux
+  user:
+    name: "{{ users.linux.username }}"
+    state: present
+    groups: "{{ users.linux.groups }}"
```
Comment (qburst-praven) [inline ansible/user-management/tasks/linux-add-user.yml:15]: we should check for the user and create if it doesnt already exist.

File: ansible/user-management/tasks/linux-add-user.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    password: "{{ users.linux.password }}"
+    shell: "{{ shell | default('/bin/bash') }}"
+    comment: "{{ comment | default('') }}"
+    update_password: always
+  when: "ansible_os_family == 'RedHat' or ansible_os_family == 'Debian' and user_check.rc != 0"
+
```
Comment (qburst-praven) [inline ansible/user-management/tasks/linux-add-user.yml:21]: pls add an option to add the SSH Key if provided

File: ansible/user-management/tasks/aws-add-user.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
+---
+- name: Ensuring AWS IAM User does not exist
+  amazon.aws.iam_user_info:
+    name: "{{ users.aws.username }}"
```
Comment (qburst-praven) [inline ansible/user-management/tasks/aws-add-user.yml:4]: this is going to be common for all 5 tasks. Please add an option to give the users as a list and iterate over the same. As is I can only have one user per task unless I am mistaken.

**Review by qburst-praven — CHANGES_REQUESTED** [review]

File: ansible/user-management/tasks/aws-add-user.yml (hunk 7 above)
Comment (qburst-arjunm) [reply ansible/user-management/tasks/aws-add-user.yml]: Added loop for the same and now multiple users can be created using the list.

File: ansible/user-management/tasks/linux-add-user.yml (hunk 6 above)
Comment (qburst-arjunm) [reply ansible/user-management/tasks/linux-add-user.yml:18]: Adding an option to copy the ssh key to the authorized_key section as discussed and modify uid of the user if the user already exists.

File: ansible/user-management/tasks/linux-add-user.yml (hunk 5 above)
Comment (qburst-arjunm) [reply ansible/user-management/tasks/linux-add-user.yml]: The job will update the uid if the user already exists.

File: ansible/user-management/tasks/linux-add-user.yml (hunk 4 above)
Comment (qburst-arjunm) [reply ansible/user-management/tasks/linux-add-user.yml]: Removing the password section and adding an option to copy the ssh pub key to the authorized key section.

File: ansible/README.md (hunk 3 above)
Comment (qburst-arjunm) [reply ansible/README.md:15]: This is updated.

File: ansible/user-management/README.md (hunk 2 above)
Comment (qburst-arjunm) [reply ansible/user-management/README.md]: This is updated.

File: README.md (hunk 1 above)
Comment (qburst-arjunm) [reply README.md]: Updated as mentioned.

**Review by qburst-praven — APPROVED** [review]

**Pull request merged**